.github/workflows/build_and_test.yml

```diff
@@ -131,16 +131,16 @@ jobs:
       # PyArrow is not supported in PyPy yet, see ARROW-2651.
       # TODO(SPARK-32247): scipy installation with PyPy fails for an unknown reason.
       run: |
-        python3.6 -m pip install numpy 'pyarrow<3.0.0' pandas scipy xmlrunner
+        python3.6 -m pip install numpy 'pyarrow<3.0.0' 'pandas<1.1.1' scipy xmlrunner
```

Comment:
Sure, either setting an upper bound or porting https://github.com/apache/spark/commit/07a9885f2792be1353f4a923d649e90bc431cb38 back should be fine. We're picking one version to test anyway.

BTW, I think `DataFrame.from_item` was removed in pandas 1.0.0 (https://github.com/pandas-dev/pandas/blob/master/doc/source/whatsnew/v1.0.0.rst#removal-of-prior-version-deprecationschanges).

Reply:
Got it. In that case, could you backport 07a9885 ?

Reply:
Sure, no problem.